docs/testing/integration.md

```diff
@@ -0,0 +1,330 @@
+        options.DatabaseSchemaName = martenSettings.SchemaName;
+    }
+
+    return options;
+}).UseLightweightSessions();
+```
+<sup><a href='https://github.com/JasperFx/marten/blob/master/src/IssueService/Startup.cs#L32-L47' title='Snippet source file'>snippet source</a> | <a href='#snippet-sample_integration_use_scheme_name' title='Start of snippet'>anchor</a></sup>
+<!-- endSnippet -->
+
+::: warning
+Keep note that Marten can be configured to generate static code on startup that contains the scheme name, so it could be beneficial to turn that off for your integration tests:
+:::
+
+```cs
+var martenSettings = configuration.GetSection(MartenSettings.SECTION).Get<MartenSettings>() ?? new MartenSettings();
+if (martenSettings.FromTests)
+{
+    martenConfiguration.ApplyAllDatabaseChangesOnStartup();
+}
+else
+{
+    martenConfiguration.OptimizeArtifactWorkflow(TypeLoadMode.Static);
+}
+```
+
+## Integrating with Wolverine
+
+Whenever wolverine's messaging is used within your application, actions may be delayed. Luckily there is a method to await Wolverine processing like this:
+
+```cs
+// This method allows us to make HTTP calls into our system
+// in memory with Alba, but do so within Wolverine's test support
+// for message tracking to both record outgoing messages and to ensure
+// that any cascaded work spawned by the initial command is completed
+// before passing control back to the calling test
+protected async Task<(ITrackedSession, IScenarioResult)> TrackedHttpCall(Action<Scenario> configuration)
+{
+    IScenarioResult result = null;
+     
+    // The outer part is tying into Wolverine's test support
+    // to "wait" for all detected message activity to complete
+    var tracked = await Host.ExecuteAndWaitAsync(async () =>
+    {
+        // The inner part here is actually making an HTTP request
+        // to the system under test with Alba
+        result = await Host.Scenario(configuration);
+    });
+ 
+    return (tracked, result);
+}
+```
+
+Just add above to your `IntegrationContext` class and you'll be able to execute wolverine endpoints like: 
+
+```cs
+var (tracked, _) = await TrackedHttpCall(x =>
+{
+    // Send a JSON post with the DebitAccount command through the HTTP endpoint
+    // BUT, it's all running in process
+    x.Post.Json(new WithdrawFromAccount(account.Id, 1300)).ToUrl("/accounts/debit");
+
+    // This is the default behavior anyway, but still good to show it here
+    x.StatusCodeShouldBeOk();
+});
+
+// And also assert that an AccountUpdated message was published as well
+var updated = tracked.Sent.SingleMessage<AccountUpdated>();
+updated.AccountId.ShouldBe(account.Id);
+updated.Balance.ShouldBe(1300);
+```
+
+Furthermore it could be beneficial to disable all external wolverine transports to test in isolation, by adding this to the Alba host setup:
+
+```cs
+services.DisableAllExternalWolverineTransports();
+```
+
+## Testing Event Projections
```

Comment:
Personally, I'd recommend testing event projections also through the integration testing. The method below can also be a bit dangerous, as you cannot run that in parallel (unless you use a different database or lockid and distinct set of projections).

Reply:
Yeah, in my own code I use this. It is amazingly stable, but I did not want to put semaphores in the documentation 😆 .

If you point me to the right direction using a lock id, I will try it out and use that. Would you keep all events in one scheme and all projections in another? Or do you split the projections by lockid?

```cs
private readonly SemaphoreSlim _daemonInACage = new(1, 1);
/// <summary>
/// 1. Start generation of projections
/// 2. Wait for projections to be projected
/// </summary>
public async Task GenerateProjectionsAsync<TView>(CancellationToken cancellationToken)
{
    await _daemonInACage.WaitAsync(30.Seconds(), cancellationToken);
    try
    {
        using var daemon = await Store.BuildProjectionDaemonAsync();
        await daemon.StartAllShards();
        await daemon.RebuildProjection<TView>(10.Seconds(), cancellationToken);
    }
    finally
    {
        _daemonInACage.Release();
    }
}
```

Reply:
For now I added a TODO, since documentation on most parts of integration testing is better then complete I suppose